tests/berlin/eip2929_gas_cost_increases/test_gas_cost.py

```diff
@@ -0,0 +1,1071 @@
+    env = Environment()
+
+    account = pre.deploy_contract(
+        initial_setup_bytecode(pre.fund_eoa(amount=0)) + contract_bytecode,
+        storage={0: 0xDEADBEEF},
+    )
+
+    tx = Transaction(
+        to=account,
+        sender=pre.fund_eoa(),
+        gas_limit=1_000_000,
+        protected=True,
+    )
+
+    post = {account: Account(storage={0: 0x00})}
+
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize(
+    "contract_bytecode",
+    [
+        pytest.param(
+            {
+                "address": 0x01,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_1_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x02,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_2_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x03,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_3_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x04,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_4_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x05,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_5_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x06,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_6_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x07,
+                "value": 1,
+                "args_offset": 0,
+                "args_size": 0,
+                "output_offset": 0,
+                "output_size": 1,
+            },
+            id="call_address_7_params_value_1_output_size_1",
+        ),
+        pytest.param(
+            {
+                "address": 0x08,
+                "value": 1,
```

Comment:
so looks like the parametrization in all this tests really is 

adress, value, output size. 

it can be one test with that many parametrizations 

for each addesss
  for each value
    for each output size  

and so on. if the test body does not change.

Reply:
Had issues with parametrizing with expected values that depend on specific inputs. Ended up changing these to parametrize the inputs and created separate tests for the permutations of the expected output.

I noticed the expected gas cost for many cases are different from the original tests. I'm looking at those closely to figure out the actual values to use, there may be additional changes to the contract based on certain scenarios. I'll check dr to figure out the differences there.

Does the current approach seem reasonable? If so, I plan to add the remaining test cases that use `CALLCODE`, `DELEGATECALL`, etc.